src/data/nat/factorial.lean

```diff
@@ -143,4 +143,37 @@ begin
   { exact add_factorial_succ_le_factorial_add_succ i h }
 end
 
+section desc_fac
+
+/-- desc_fac n k = (n + k)! / n!, but implemented in a recursive way for calculation.
+This is closely related to `ring_theory.polynomial.pochhammer`, but much less general. -/
+def desc_fac (n : ℕ) : ℕ → ℕ
+| 0 := 1
+| (k + 1) := (n + k + 1) * desc_fac k
+
+@[simp] lemma desc_fac_zero (n : ℕ) : desc_fac n 0 = 1 := rfl
+
+@[simp] lemma zero_desc_fac (k : ℕ) : desc_fac 0 k = k! :=
+begin
+  induction k with t ht, refl,
+  unfold desc_fac, rw [ht, zero_add, nat.factorial_succ]
+end
+
+lemma desc_fac_succ {n k : ℕ} : desc_fac n k.succ = (n + k + 1) * desc_fac n k := rfl
+
+lemma succ_desc_fac {n k : ℕ} : (n + 1) * desc_fac n.succ k = (n + k + 1) * desc_fac n k :=
+begin
+  induction k with t ht, simp!, rw desc_fac_succ, rw desc_fac_succ,
```

Comment:
The `simp` disposes of the base case on its own, right? Meaning the `rw desc_fac_succ`s are applied to the second goal? If so they should go on a separate line (or you can combine them with the `rw`s on the last line).

Reply:
yep, for sure!